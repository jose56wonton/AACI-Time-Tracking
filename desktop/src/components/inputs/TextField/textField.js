import React from 'react';
import PropTypes from 'prop-types';

import { TextField as MUTextField } from '@material-ui/core';
import {getIn} from 'formik'

const TextField = ({ field, form, label, className, type, margin, helper }) => (
  <MUTextField
    {...field}
    label={label}
    type={type}
    InputLabelProps={{ shrink: true }}
    fullWidth
    className={className}
    margin={margin}
    FormHelperTextProps={helper ==='none' ? { style:{ display:'none' },error: true } : { error: true  }} 
    helperText={getIn(form.errors, field.name)}
  />
);

TextField.defaultProps = {
  type: 'search',
  margin: 'normal',
  helper: 'normal'
};

TextField.propTypes = {
  field: PropTypes.object.isRequired,
  form: PropTypes.object.isRequired,
  value: PropTypes.string,
  onChange: PropTypes.func,
  type: PropTypes.string,
  label: PropTypes.string.isRequired,
  margin: PropTypes.string,
  helper: PropTypes.oneOf(['none','normal'])
};

export default TextField;
