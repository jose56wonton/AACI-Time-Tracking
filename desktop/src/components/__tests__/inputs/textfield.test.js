import React from 'react';
import { mount } from 'enzyme';

import { TextField } from 'components/inputs/TextField/textField';
import TextFieldHOC from 'components/inputs/TextField';

const props =  {  
  form:{
    errors: {}
  },
  field: {
    name: 'name'
  },
  label: 'label',
  labelProps: {},
  formControlProps: {},
  margin: 'normal',
  classes: {},
  helper: 'normal',
  fullWidth: true
};

const setup = overRides => {  
  return mount(<TextField {...props} {...overRides}/>);    
};

const setupHOC = overRides => {
  return mount(<TextFieldHOC {...props} {...overRides}/>);
};

describe('TextField Input', () => {
  it('should render correctly', () => {
    const wrapper = setup();
    expect(wrapper).toMatchSnapshot();    
  });
  it('should render correctly withStyles', () => {
    const wrapper = setupHOC();
    expect(wrapper).toMatchSnapshot();   
  }); 
  it('should render no helper text when helper="none"', () => {
    const wrapper = setup({ helper: 'none' });
    expect(wrapper).toMatchSnapshot();    
  });
});