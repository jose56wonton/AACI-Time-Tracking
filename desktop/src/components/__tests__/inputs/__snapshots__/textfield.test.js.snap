// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextField Input should render correctly 1`] = `
<Component
  classes={Object {}}
  field={
    Object {
      "name": "name",
    }
  }
  form={
    Object {
      "errors": Object {},
    }
  }
  formControlProps={Object {}}
  fullWidth={true}
  helper="normal"
  label="label"
  labelProps={Object {}}
  margin="normal"
  type="search"
>
  <TextField
    FormHelperTextProps={
      Object {
        "error": true,
      }
    }
    InputLabelProps={
      Object {
        "shrink": true,
      }
    }
    fullWidth={true}
    helperText=" "
    label="label"
    margin="normal"
    name="name"
    required={false}
    select={false}
    type="search"
    variant="standard"
  >
    <WithStyles(FormControl)
      fullWidth={true}
      margin="normal"
      required={false}
      variant="standard"
    >
      <FormControl
        classes={
          Object {
            "fullWidth": "MuiFormControl-fullWidth-4",
            "marginDense": "MuiFormControl-marginDense-3",
            "marginNormal": "MuiFormControl-marginNormal-2",
            "root": "MuiFormControl-root-1",
          }
        }
        component="div"
        disabled={false}
        error={false}
        fullWidth={true}
        margin="normal"
        required={false}
        variant="standard"
      >
        <div
          className="MuiFormControl-root-1 MuiFormControl-marginNormal-2 MuiFormControl-fullWidth-4"
        >
          <WithStyles(InputLabel)
            shrink={true}
          >
            <InputLabel
              classes={
                Object {
                  "animated": "MuiInputLabel-animated-9",
                  "filled": "MuiInputLabel-filled-10",
                  "formControl": "MuiInputLabel-formControl-6",
                  "marginDense": "MuiInputLabel-marginDense-7",
                  "outlined": "MuiInputLabel-outlined-11",
                  "root": "MuiInputLabel-root-5",
                  "shrink": "MuiInputLabel-shrink-8",
                }
              }
              disableAnimation={false}
              shrink={true}
            >
              <WithStyles(FormLabel)
                className="MuiInputLabel-root-5 MuiInputLabel-formControl-6 MuiInputLabel-animated-9 MuiInputLabel-shrink-8"
                data-shrink={true}
              >
                <FormLabel
                  className="MuiInputLabel-root-5 MuiInputLabel-formControl-6 MuiInputLabel-animated-9 MuiInputLabel-shrink-8"
                  classes={
                    Object {
                      "asterisk": "MuiFormLabel-asterisk-18",
                      "disabled": "MuiFormLabel-disabled-14",
                      "error": "MuiFormLabel-error-15",
                      "filled": "MuiFormLabel-filled-16",
                      "focused": "MuiFormLabel-focused-13",
                      "required": "MuiFormLabel-required-17",
                      "root": "MuiFormLabel-root-12",
                    }
                  }
                  component="label"
                  data-shrink={true}
                >
                  <label
                    className="MuiFormLabel-root-12 MuiInputLabel-root-5 MuiInputLabel-formControl-6 MuiInputLabel-animated-9 MuiInputLabel-shrink-8"
                    data-shrink={true}
                  >
                    label
                  </label>
                </FormLabel>
              </WithStyles(FormLabel)>
            </InputLabel>
          </WithStyles(InputLabel)>
          <WithStyles(Input)
            fullWidth={true}
            name="name"
            type="search"
          >
            <Input
              classes={
                Object {
                  "disabled": "MuiInput-disabled-22",
                  "error": "MuiInput-error-24",
                  "focused": "MuiInput-focused-21",
                  "formControl": "MuiInput-formControl-20",
                  "fullWidth": "MuiInput-fullWidth-26",
                  "input": "MuiInput-input-27",
                  "inputMarginDense": "MuiInput-inputMarginDense-28",
                  "inputMultiline": "MuiInput-inputMultiline-29",
                  "inputType": "MuiInput-inputType-30",
                  "inputTypeSearch": "MuiInput-inputTypeSearch-31",
                  "multiline": "MuiInput-multiline-25",
                  "root": "MuiInput-root-19",
                  "underline": "MuiInput-underline-23",
                }
              }
              fullWidth={true}
              name="name"
              type="search"
            >
              <WithStyles(InputBase)
                classes={
                  Object {
                    "disabled": "MuiInput-disabled-22",
                    "error": "MuiInput-error-24",
                    "focused": "MuiInput-focused-21",
                    "formControl": "MuiInput-formControl-20",
                    "fullWidth": "MuiInput-fullWidth-26",
                    "input": "MuiInput-input-27",
                    "inputMarginDense": "MuiInput-inputMarginDense-28",
                    "inputMultiline": "MuiInput-inputMultiline-29",
                    "inputType": "MuiInput-inputType-30",
                    "inputTypeSearch": "MuiInput-inputTypeSearch-31",
                    "multiline": "MuiInput-multiline-25",
                    "root": "MuiInput-root-19 MuiInput-underline-23",
                    "underline": null,
                  }
                }
                fullWidth={true}
                inputComponent="input"
                multiline={false}
                name="name"
                type="search"
              >
                <InputBase
                  classes={
                    Object {
                      "adornedEnd": "MuiInputBase-adornedEnd-37",
                      "adornedStart": "MuiInputBase-adornedStart-36",
                      "disabled": "MuiInputBase-disabled-35 MuiInput-disabled-22",
                      "error": "MuiInputBase-error-38 MuiInput-error-24",
                      "focused": "MuiInputBase-focused-34 MuiInput-focused-21",
                      "formControl": "MuiInputBase-formControl-33 MuiInput-formControl-20",
                      "fullWidth": "MuiInputBase-fullWidth-41 MuiInput-fullWidth-26",
                      "input": "MuiInputBase-input-42 MuiInput-input-27",
                      "inputAdornedEnd": "MuiInputBase-inputAdornedEnd-48",
                      "inputAdornedStart": "MuiInputBase-inputAdornedStart-47",
                      "inputMarginDense": "MuiInputBase-inputMarginDense-43 MuiInput-inputMarginDense-28",
                      "inputMultiline": "MuiInputBase-inputMultiline-44 MuiInput-inputMultiline-29",
                      "inputType": "MuiInputBase-inputType-45 MuiInput-inputType-30",
                      "inputTypeSearch": "MuiInputBase-inputTypeSearch-46 MuiInput-inputTypeSearch-31",
                      "marginDense": "MuiInputBase-marginDense-39",
                      "multiline": "MuiInputBase-multiline-40 MuiInput-multiline-25",
                      "root": "MuiInputBase-root-32 MuiInput-root-19 MuiInput-underline-23",
                    }
                  }
                  fullWidth={true}
                  inputComponent="input"
                  multiline={false}
                  name="name"
                  type="search"
                >
                  <div
                    className="MuiInputBase-root-32 MuiInput-root-19 MuiInput-underline-23 MuiInputBase-fullWidth-41 MuiInput-fullWidth-26 MuiInputBase-formControl-33 MuiInput-formControl-20"
                    onClick={[Function]}
                  >
                    <input
                      aria-invalid={false}
                      className="MuiInputBase-input-42 MuiInput-input-27 MuiInputBase-inputType-45 MuiInput-inputType-30 MuiInputBase-inputTypeSearch-46 MuiInput-inputTypeSearch-31"
                      disabled={false}
                      name="name"
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      required={false}
                      type="search"
                    />
                  </div>
                </InputBase>
              </WithStyles(InputBase)>
            </Input>
          </WithStyles(Input)>
          <WithStyles(FormHelperText)
            error={true}
          >
            <FormHelperText
              classes={
                Object {
                  "contained": "MuiFormHelperText-contained-53",
                  "disabled": "MuiFormHelperText-disabled-51",
                  "error": "MuiFormHelperText-error-50",
                  "filled": "MuiFormHelperText-filled-55",
                  "focused": "MuiFormHelperText-focused-54",
                  "marginDense": "MuiFormHelperText-marginDense-52",
                  "required": "MuiFormHelperText-required-56",
                  "root": "MuiFormHelperText-root-49",
                }
              }
              component="p"
              error={true}
            >
              <p
                className="MuiFormHelperText-root-49 MuiFormHelperText-error-50"
              >
                 
              </p>
            </FormHelperText>
          </WithStyles(FormHelperText)>
        </div>
      </FormControl>
    </WithStyles(FormControl)>
  </TextField>
</Component>
`;

exports[`TextField Input should render correctly withStyles 1`] = `
<Component
  classes={Object {}}
  field={
    Object {
      "name": "name",
    }
  }
  form={
    Object {
      "errors": Object {},
    }
  }
  formControlProps={Object {}}
  fullWidth={true}
  helper="normal"
  label="label"
  labelProps={Object {}}
  margin="normal"
  type="search"
>
  <TextField
    FormHelperTextProps={
      Object {
        "error": true,
      }
    }
    InputLabelProps={
      Object {
        "shrink": true,
      }
    }
    fullWidth={true}
    helperText=" "
    label="label"
    margin="normal"
    name="name"
    required={false}
    select={false}
    type="search"
    variant="standard"
  >
    <WithStyles(FormControl)
      fullWidth={true}
      margin="normal"
      required={false}
      variant="standard"
    >
      <FormControl
        classes={
          Object {
            "fullWidth": "MuiFormControl-fullWidth-4",
            "marginDense": "MuiFormControl-marginDense-3",
            "marginNormal": "MuiFormControl-marginNormal-2",
            "root": "MuiFormControl-root-1",
          }
        }
        component="div"
        disabled={false}
        error={false}
        fullWidth={true}
        margin="normal"
        required={false}
        variant="standard"
      >
        <div
          className="MuiFormControl-root-1 MuiFormControl-marginNormal-2 MuiFormControl-fullWidth-4"
        >
          <WithStyles(InputLabel)
            shrink={true}
          >
            <InputLabel
              classes={
                Object {
                  "animated": "MuiInputLabel-animated-9",
                  "filled": "MuiInputLabel-filled-10",
                  "formControl": "MuiInputLabel-formControl-6",
                  "marginDense": "MuiInputLabel-marginDense-7",
                  "outlined": "MuiInputLabel-outlined-11",
                  "root": "MuiInputLabel-root-5",
                  "shrink": "MuiInputLabel-shrink-8",
                }
              }
              disableAnimation={false}
              shrink={true}
            >
              <WithStyles(FormLabel)
                className="MuiInputLabel-root-5 MuiInputLabel-formControl-6 MuiInputLabel-animated-9 MuiInputLabel-shrink-8"
                data-shrink={true}
              >
                <FormLabel
                  className="MuiInputLabel-root-5 MuiInputLabel-formControl-6 MuiInputLabel-animated-9 MuiInputLabel-shrink-8"
                  classes={
                    Object {
                      "asterisk": "MuiFormLabel-asterisk-18",
                      "disabled": "MuiFormLabel-disabled-14",
                      "error": "MuiFormLabel-error-15",
                      "filled": "MuiFormLabel-filled-16",
                      "focused": "MuiFormLabel-focused-13",
                      "required": "MuiFormLabel-required-17",
                      "root": "MuiFormLabel-root-12",
                    }
                  }
                  component="label"
                  data-shrink={true}
                >
                  <label
                    className="MuiFormLabel-root-12 MuiInputLabel-root-5 MuiInputLabel-formControl-6 MuiInputLabel-animated-9 MuiInputLabel-shrink-8"
                    data-shrink={true}
                  >
                    label
                  </label>
                </FormLabel>
              </WithStyles(FormLabel)>
            </InputLabel>
          </WithStyles(InputLabel)>
          <WithStyles(Input)
            fullWidth={true}
            name="name"
            type="search"
          >
            <Input
              classes={
                Object {
                  "disabled": "MuiInput-disabled-22",
                  "error": "MuiInput-error-24",
                  "focused": "MuiInput-focused-21",
                  "formControl": "MuiInput-formControl-20",
                  "fullWidth": "MuiInput-fullWidth-26",
                  "input": "MuiInput-input-27",
                  "inputMarginDense": "MuiInput-inputMarginDense-28",
                  "inputMultiline": "MuiInput-inputMultiline-29",
                  "inputType": "MuiInput-inputType-30",
                  "inputTypeSearch": "MuiInput-inputTypeSearch-31",
                  "multiline": "MuiInput-multiline-25",
                  "root": "MuiInput-root-19",
                  "underline": "MuiInput-underline-23",
                }
              }
              fullWidth={true}
              name="name"
              type="search"
            >
              <WithStyles(InputBase)
                classes={
                  Object {
                    "disabled": "MuiInput-disabled-22",
                    "error": "MuiInput-error-24",
                    "focused": "MuiInput-focused-21",
                    "formControl": "MuiInput-formControl-20",
                    "fullWidth": "MuiInput-fullWidth-26",
                    "input": "MuiInput-input-27",
                    "inputMarginDense": "MuiInput-inputMarginDense-28",
                    "inputMultiline": "MuiInput-inputMultiline-29",
                    "inputType": "MuiInput-inputType-30",
                    "inputTypeSearch": "MuiInput-inputTypeSearch-31",
                    "multiline": "MuiInput-multiline-25",
                    "root": "MuiInput-root-19 MuiInput-underline-23",
                    "underline": null,
                  }
                }
                fullWidth={true}
                inputComponent="input"
                multiline={false}
                name="name"
                type="search"
              >
                <InputBase
                  classes={
                    Object {
                      "adornedEnd": "MuiInputBase-adornedEnd-37",
                      "adornedStart": "MuiInputBase-adornedStart-36",
                      "disabled": "MuiInputBase-disabled-35 MuiInput-disabled-22",
                      "error": "MuiInputBase-error-38 MuiInput-error-24",
                      "focused": "MuiInputBase-focused-34 MuiInput-focused-21",
                      "formControl": "MuiInputBase-formControl-33 MuiInput-formControl-20",
                      "fullWidth": "MuiInputBase-fullWidth-41 MuiInput-fullWidth-26",
                      "input": "MuiInputBase-input-42 MuiInput-input-27",
                      "inputAdornedEnd": "MuiInputBase-inputAdornedEnd-48",
                      "inputAdornedStart": "MuiInputBase-inputAdornedStart-47",
                      "inputMarginDense": "MuiInputBase-inputMarginDense-43 MuiInput-inputMarginDense-28",
                      "inputMultiline": "MuiInputBase-inputMultiline-44 MuiInput-inputMultiline-29",
                      "inputType": "MuiInputBase-inputType-45 MuiInput-inputType-30",
                      "inputTypeSearch": "MuiInputBase-inputTypeSearch-46 MuiInput-inputTypeSearch-31",
                      "marginDense": "MuiInputBase-marginDense-39",
                      "multiline": "MuiInputBase-multiline-40 MuiInput-multiline-25",
                      "root": "MuiInputBase-root-32 MuiInput-root-19 MuiInput-underline-23",
                    }
                  }
                  fullWidth={true}
                  inputComponent="input"
                  multiline={false}
                  name="name"
                  type="search"
                >
                  <div
                    className="MuiInputBase-root-32 MuiInput-root-19 MuiInput-underline-23 MuiInputBase-fullWidth-41 MuiInput-fullWidth-26 MuiInputBase-formControl-33 MuiInput-formControl-20"
                    onClick={[Function]}
                  >
                    <input
                      aria-invalid={false}
                      className="MuiInputBase-input-42 MuiInput-input-27 MuiInputBase-inputType-45 MuiInput-inputType-30 MuiInputBase-inputTypeSearch-46 MuiInput-inputTypeSearch-31"
                      disabled={false}
                      name="name"
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      required={false}
                      type="search"
                    />
                  </div>
                </InputBase>
              </WithStyles(InputBase)>
            </Input>
          </WithStyles(Input)>
          <WithStyles(FormHelperText)
            error={true}
          >
            <FormHelperText
              classes={
                Object {
                  "contained": "MuiFormHelperText-contained-53",
                  "disabled": "MuiFormHelperText-disabled-51",
                  "error": "MuiFormHelperText-error-50",
                  "filled": "MuiFormHelperText-filled-55",
                  "focused": "MuiFormHelperText-focused-54",
                  "marginDense": "MuiFormHelperText-marginDense-52",
                  "required": "MuiFormHelperText-required-56",
                  "root": "MuiFormHelperText-root-49",
                }
              }
              component="p"
              error={true}
            >
              <p
                className="MuiFormHelperText-root-49 MuiFormHelperText-error-50"
              >
                 
              </p>
            </FormHelperText>
          </WithStyles(FormHelperText)>
        </div>
      </FormControl>
    </WithStyles(FormControl)>
  </TextField>
</Component>
`;

exports[`TextField Input should render no helper text when helper="none" 1`] = `
<Component
  classes={Object {}}
  field={
    Object {
      "name": "name",
    }
  }
  form={
    Object {
      "errors": Object {},
    }
  }
  formControlProps={Object {}}
  fullWidth={true}
  helper="none"
  label="label"
  labelProps={Object {}}
  margin="normal"
  type="search"
>
  <TextField
    FormHelperTextProps={
      Object {
        "error": true,
        "style": Object {
          "display": "none",
        },
      }
    }
    InputLabelProps={
      Object {
        "shrink": true,
      }
    }
    fullWidth={true}
    helperText=" "
    label="label"
    margin="normal"
    name="name"
    required={false}
    select={false}
    type="search"
    variant="standard"
  >
    <WithStyles(FormControl)
      fullWidth={true}
      margin="normal"
      required={false}
      variant="standard"
    >
      <FormControl
        classes={
          Object {
            "fullWidth": "MuiFormControl-fullWidth-4",
            "marginDense": "MuiFormControl-marginDense-3",
            "marginNormal": "MuiFormControl-marginNormal-2",
            "root": "MuiFormControl-root-1",
          }
        }
        component="div"
        disabled={false}
        error={false}
        fullWidth={true}
        margin="normal"
        required={false}
        variant="standard"
      >
        <div
          className="MuiFormControl-root-1 MuiFormControl-marginNormal-2 MuiFormControl-fullWidth-4"
        >
          <WithStyles(InputLabel)
            shrink={true}
          >
            <InputLabel
              classes={
                Object {
                  "animated": "MuiInputLabel-animated-9",
                  "filled": "MuiInputLabel-filled-10",
                  "formControl": "MuiInputLabel-formControl-6",
                  "marginDense": "MuiInputLabel-marginDense-7",
                  "outlined": "MuiInputLabel-outlined-11",
                  "root": "MuiInputLabel-root-5",
                  "shrink": "MuiInputLabel-shrink-8",
                }
              }
              disableAnimation={false}
              shrink={true}
            >
              <WithStyles(FormLabel)
                className="MuiInputLabel-root-5 MuiInputLabel-formControl-6 MuiInputLabel-animated-9 MuiInputLabel-shrink-8"
                data-shrink={true}
              >
                <FormLabel
                  className="MuiInputLabel-root-5 MuiInputLabel-formControl-6 MuiInputLabel-animated-9 MuiInputLabel-shrink-8"
                  classes={
                    Object {
                      "asterisk": "MuiFormLabel-asterisk-18",
                      "disabled": "MuiFormLabel-disabled-14",
                      "error": "MuiFormLabel-error-15",
                      "filled": "MuiFormLabel-filled-16",
                      "focused": "MuiFormLabel-focused-13",
                      "required": "MuiFormLabel-required-17",
                      "root": "MuiFormLabel-root-12",
                    }
                  }
                  component="label"
                  data-shrink={true}
                >
                  <label
                    className="MuiFormLabel-root-12 MuiInputLabel-root-5 MuiInputLabel-formControl-6 MuiInputLabel-animated-9 MuiInputLabel-shrink-8"
                    data-shrink={true}
                  >
                    label
                  </label>
                </FormLabel>
              </WithStyles(FormLabel)>
            </InputLabel>
          </WithStyles(InputLabel)>
          <WithStyles(Input)
            fullWidth={true}
            name="name"
            type="search"
          >
            <Input
              classes={
                Object {
                  "disabled": "MuiInput-disabled-22",
                  "error": "MuiInput-error-24",
                  "focused": "MuiInput-focused-21",
                  "formControl": "MuiInput-formControl-20",
                  "fullWidth": "MuiInput-fullWidth-26",
                  "input": "MuiInput-input-27",
                  "inputMarginDense": "MuiInput-inputMarginDense-28",
                  "inputMultiline": "MuiInput-inputMultiline-29",
                  "inputType": "MuiInput-inputType-30",
                  "inputTypeSearch": "MuiInput-inputTypeSearch-31",
                  "multiline": "MuiInput-multiline-25",
                  "root": "MuiInput-root-19",
                  "underline": "MuiInput-underline-23",
                }
              }
              fullWidth={true}
              name="name"
              type="search"
            >
              <WithStyles(InputBase)
                classes={
                  Object {
                    "disabled": "MuiInput-disabled-22",
                    "error": "MuiInput-error-24",
                    "focused": "MuiInput-focused-21",
                    "formControl": "MuiInput-formControl-20",
                    "fullWidth": "MuiInput-fullWidth-26",
                    "input": "MuiInput-input-27",
                    "inputMarginDense": "MuiInput-inputMarginDense-28",
                    "inputMultiline": "MuiInput-inputMultiline-29",
                    "inputType": "MuiInput-inputType-30",
                    "inputTypeSearch": "MuiInput-inputTypeSearch-31",
                    "multiline": "MuiInput-multiline-25",
                    "root": "MuiInput-root-19 MuiInput-underline-23",
                    "underline": null,
                  }
                }
                fullWidth={true}
                inputComponent="input"
                multiline={false}
                name="name"
                type="search"
              >
                <InputBase
                  classes={
                    Object {
                      "adornedEnd": "MuiInputBase-adornedEnd-37",
                      "adornedStart": "MuiInputBase-adornedStart-36",
                      "disabled": "MuiInputBase-disabled-35 MuiInput-disabled-22",
                      "error": "MuiInputBase-error-38 MuiInput-error-24",
                      "focused": "MuiInputBase-focused-34 MuiInput-focused-21",
                      "formControl": "MuiInputBase-formControl-33 MuiInput-formControl-20",
                      "fullWidth": "MuiInputBase-fullWidth-41 MuiInput-fullWidth-26",
                      "input": "MuiInputBase-input-42 MuiInput-input-27",
                      "inputAdornedEnd": "MuiInputBase-inputAdornedEnd-48",
                      "inputAdornedStart": "MuiInputBase-inputAdornedStart-47",
                      "inputMarginDense": "MuiInputBase-inputMarginDense-43 MuiInput-inputMarginDense-28",
                      "inputMultiline": "MuiInputBase-inputMultiline-44 MuiInput-inputMultiline-29",
                      "inputType": "MuiInputBase-inputType-45 MuiInput-inputType-30",
                      "inputTypeSearch": "MuiInputBase-inputTypeSearch-46 MuiInput-inputTypeSearch-31",
                      "marginDense": "MuiInputBase-marginDense-39",
                      "multiline": "MuiInputBase-multiline-40 MuiInput-multiline-25",
                      "root": "MuiInputBase-root-32 MuiInput-root-19 MuiInput-underline-23",
                    }
                  }
                  fullWidth={true}
                  inputComponent="input"
                  multiline={false}
                  name="name"
                  type="search"
                >
                  <div
                    className="MuiInputBase-root-32 MuiInput-root-19 MuiInput-underline-23 MuiInputBase-fullWidth-41 MuiInput-fullWidth-26 MuiInputBase-formControl-33 MuiInput-formControl-20"
                    onClick={[Function]}
                  >
                    <input
                      aria-invalid={false}
                      className="MuiInputBase-input-42 MuiInput-input-27 MuiInputBase-inputType-45 MuiInput-inputType-30 MuiInputBase-inputTypeSearch-46 MuiInput-inputTypeSearch-31"
                      disabled={false}
                      name="name"
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      required={false}
                      type="search"
                    />
                  </div>
                </InputBase>
              </WithStyles(InputBase)>
            </Input>
          </WithStyles(Input)>
          <WithStyles(FormHelperText)
            error={true}
            style={
              Object {
                "display": "none",
              }
            }
          >
            <FormHelperText
              classes={
                Object {
                  "contained": "MuiFormHelperText-contained-53",
                  "disabled": "MuiFormHelperText-disabled-51",
                  "error": "MuiFormHelperText-error-50",
                  "filled": "MuiFormHelperText-filled-55",
                  "focused": "MuiFormHelperText-focused-54",
                  "marginDense": "MuiFormHelperText-marginDense-52",
                  "required": "MuiFormHelperText-required-56",
                  "root": "MuiFormHelperText-root-49",
                }
              }
              component="p"
              error={true}
              style={
                Object {
                  "display": "none",
                }
              }
            >
              <p
                className="MuiFormHelperText-root-49 MuiFormHelperText-error-50"
                style={
                  Object {
                    "display": "none",
                  }
                }
              >
                 
              </p>
            </FormHelperText>
          </WithStyles(FormHelperText)>
        </div>
      </FormControl>
    </WithStyles(FormControl)>
  </TextField>
</Component>
`;
